realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -40,34 +42,41 @@
 public class PermissionManager implements Closeable {
 
     // Reference counted cache equivalent to how Realm instances work.
-    private static ThreadLocal<PermissionManager> permissionManager = new ThreadLocal<PermissionManager>() {
-        @Override
-        protected PermissionManager initialValue() {
-            return null;
-        }
-    };
+    private static Map<String, ThreadLocal<Cache>> cache = new HashMap<>();
```

Comment:
The lock can be removed if it is `ThreadLocal<Map<String, cache>>`. But this is fine to me.